File: 30DaysOfCode/day20_sorting/sorting.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    int n; 
    scanf("%d", &n);
    int *a = malloc(sizeof(int) * n);
    for(int a_i = 0; a_i < n; a_i++){
       scanf("%d", &a[a_i]);
    }
    
    int numSwaps = 0;
    
    for (int i = 0; i < n; i++) {
        int maxIndex = n - i - 1;
        for (int j = 0; j < maxIndex; j++) {
            if (a[j] > a[j + 1]) {
                int copy = a[j];
                a[j] = a[j + 1];
                a[j + 1] = copy;
                numSwaps++;
            }
        }
    }
    
    printf("Array is sorted in %d swaps.\n", numSwaps);
    printf("First Element: %d\n", a[0]);
    printf("Last Element: %d\n", a[n - 1]);

    return 0;
}

File: 30DaysOfCode/day20_sorting/sorting.php
<?php

$handle = fopen ("php://stdin", "r");
fscanf($handle, "%d",$n);
$a_temp = fgets($handle);
$a = explode(" ",$a_temp);
$a = array_map('intval', $a);

$numSwaps = 0;

for ($i = 0; $i < $n; $i++) {
    $maxIndex = $n - $i - 1;
    for ($j = 0; $j < $maxIndex; $j++) {
        if ($a[$j] > $a[$j + 1]) {
            $copy = $a[$j];
            $a[$j] = $a[$j + 1];
            $a[$j + 1] = $copy;
            $numSwaps++;
        }
    }
}

echo "Array is sorted in " . $numSwaps . " swaps.\n";
echo "First Element: " . $a[0] . "\n";
echo "Last Element: " . $a[$n - 1] . "\n";

?>
